internal/services/automation/automation_hybrid_runbook_worker_group.go

```diff
@@ -124,8 +127,10 @@ func (m HybridRunbookWorkerGroupResource) Read() sdk.ResourceFunc {
 
 			output.Name = utils.NormalizeNilableString(result.Model.Name)
 			output.AutomationAccountName = id.AutomationAccountName
-			if c := result.Model.Credential; c != nil {
-				output.CredentialName = utils.NormalizeNilableString(c.Name)
+			if c := result.Model.Properties; c != nil {
+				if c.Credential != nil {
+					output.CredentialName = pointer.From(c.Credential.Name)
+				}
```

Comment:
Can we refactor lines 123-133 to be consistent with the rest of the provider, by that I mean checking the result or response like we do in other resource for improved readability like we have [documented in the contributors docs](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/guide-new-resource.md#step-4-scaffold-an-emptynew-resource) under the example for the read function.

```suggestion
			if model := result.Model; model != nil {
				// set things here
				if props := model.Properties; props != nil {
					// set more props here
				}
			}
```

It's